### DISCUSSION TITLE
problem displaying views

### DISCUSSION THREAD

**Discussion by LeFlow** [adonisjs/core#4446] [General]
Hi

I'm starting with Adonisjs.
v6

I have a controller "test_controller.ts":
`import { HttpContext } from "@adonisjs/core/http";
export default class TestController {
  async index({ view }:HttpContext) {
    return view.render('index.edge');
  }
}`
route:
`router.get('/', [TestController, 'index'])`
the views :
layout.edge:
`<!DOCTYPE html>
<html lang="fr">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Test Adonisjs</title>
  </head>
  <body>
    <div class="container">
      @!section('body')
    </div>
  </body>
</html>`

and index.edge:
`@layout('layout')
@section('body')
  <h2>Titre</h2>
@endsection`

The problem is that when I go to the localhost:3333 page, it displays:
`@layout('layout')
@section('body')
  <h2>Titre</h2>
@endsection`

Can you help me pls.

TY

**Comment by RomainLanz** [comment]
Hey @LeFlow! 👋🏻 

`@layout` is no longer existing in the latest version of Edge.
Use components to make a layout system.

📚 https://edgejs.dev/docs/components/introduction

**Comment by LeFlow** [comment]
if I understand correctly, for this example, I must make a "header" component and a "footer" component and in my index do:
`@!component('components/header')
   Title
@!component('components/footer')`

**Reply by RomainLanz** [reply]
You have to create a layout file (e.g., `resources/views/components/layouts/app.edge`).

```html
<!DOCTYPE html>
<html lang="fr">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">

  <title>{{ title ?? 'RomainLanz' }}</title>

  {{-- Compiled Assets --}}
  @vite(['resources/js/app.js'])
</head>
<body>
  @include('partials/header')

  {{{ await $slots.main() }}}

  @include('partials/footer')
</body>
</html>
```

Then, you can use it in any of your pages.

```html
@layouts.app()
  My Page!
@end
```

**Comment by LeFlow** [comment]
Ty

I will try to do like that.

If I have a problem, I'll come back ^^

**Comment by LeFlow** [comment]
It's good but I needed to replace  `@include` by `@!component`

TY
